lua/entities/gmod_wire_expression2/core/egpfunctions.lua

```diff
@@ -1143,12 +1143,34 @@ end
 local antispam = {}
 __e2setcost(25)
 e2function void wirelink:egpHudToggle()
-	if not EGP:ValidEGP( this ) then return self:throw("Invalid wirelink!", nil) end
+	if not EGP:ValidEGP(this) then return self:throw("Invalid wirelink!", nil) end
 	if antispam[self.player] and antispam[self.player] > CurTime() then return end
 	antispam[self.player] = CurTime() + 0.1
-	umsg.Start( "EGP_HUD_Use", self.player ) umsg.Entity( this ) umsg.End()
+	net.Start("EGP_HUD_Use") net.WriteEntity(this) net.WriteInt(0, 2) net.Send(self.player)
```

Comment:
Could the toggling be server-side only and use net.WriteBool instead to enable/disable hud for client? Would be a lot cleaner and less likely for toggling to desync.

Reply:
That's a great idea. I'll look into it.